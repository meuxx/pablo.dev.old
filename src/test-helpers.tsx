import { render } from '@testing-library/react'
import { Themed } from './styles/GlobalStyle'
import React from 'react'

const renderComponent: typeof render = (ui, options) => render(ui, { wrapper: Themed, ...options })

const mockComponent = (componentName: string): React.FC => (props) => {
  const { children, ...otherProps } = props
  return (
    <div data-component={componentName} data-props={JSON.stringify(otherProps, null, ' ')}>
      {children}
    </div>
  )
}

export { renderComponent, mockComponent }
