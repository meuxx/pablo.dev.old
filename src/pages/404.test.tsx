import React from 'react'
import { mockComponent, renderComponent } from '../test-helpers'
import NotFoundPage from './404'

jest.mock('../components/Layout', () => mockComponent('Layout'))
jest.mock('gatsby-plugin-next-seo', () => ({
  GatsbySeo: mockComponent('GatsbySeo'),
}))

describe('404', () => {
  it('renders without crashing', () => {
    const { container } = renderComponent(<NotFoundPage />)

    expect(container).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
