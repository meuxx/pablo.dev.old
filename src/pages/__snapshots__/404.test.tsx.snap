// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 renders without crashing 1`] = `
.c0 {
  margin-left: 24%;
  max-width: 52em;
  padding: 4em 2em;
  width: calc(100% - 24%);
}

.c3 {
  -webkit-transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  border-bottom: dotted 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c3:hover {
  border-bottom-color: transparent;
  color: #ffffff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  color: #0B61A4;
}

.c1 {
  margin: 2em 0 0 0;
  padding: 2em 0 0 0;
}

.c1:first-child {
  margin-top: 0;
  padding-top: 0;
}

.c1 p {
  color: #b2b2b2;
  position: relative;
  margin: 0 0 2em 0;
}

.c1 p strong {
  color: #ffffff;
  font-weight: 700;
}

.c2 h1,
.c2 h2 {
  color: #ffffff;
  font-weight: 700;
}

.c2 h1 {
  margin: 0 0 1em;
  line-height: 1.5em;
}

.c2 h2 {
  margin: 0 0 1em;
  font-size: 1.5em;
  line-height: 1.5em;
}

@media screen and (max-width:1280px) {
  .c0 {
    margin-left: 30%;
    max-width: none;
    padding: 6em 3em 3em 3em;
    width: calc(100% - 30%);
  }
}

@media screen and (max-width:980px) {
  .c0 {
    margin: 0;
    padding: 6em 4em;
    width: 100%;
  }
}

@media screen and (max-width:736px) {
  .c0 {
    padding: 2.25em 1.5em 0.25em 1.5em;
  }
}

@media screen and (max-width:736px) {
  .c1 {
    margin: 1.5em 0 0 0;
    padding: 1.5em 0 0 0;
  }
}

@media screen and (max-width:736px) {
  .c2 h2 {
    font-size: 1.2em;
  }
}

<div>
  <div
    data-component="Layout"
    data-props="{}"
  >
    <div
      data-component="GatsbySeo"
      data-props="{
 \\"title\\": \\"404 Not found\\"
}"
    />
    <main
      class="c0"
    >
      <section
        class="c1"
      >
        <header
          class="c2"
        >
          <h1>
            NOT FOUND
          </h1>
        </header>
        <p>
          You just hit a route that doesn't exist... the sadness.
        </p>
        <a
          class="c3 c4"
          href="/"
        >
          Home
        </a>
      </section>
    </main>
  </div>
</div>
`;
